File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/ConsumptionStateV1.java
package com.cleverferret.v2.core.database.schema;

import com.cleverferret.v2.core.common.VersionedContract;

/**
 * Canonical position/progress row for reading/listening/watching.
 */
public record ConsumptionStateV1(
        String consumptionStateId,
        String mediaItemId,
        String mediaVariantId,
        String profileId,
        ConsumptionModeV1 mode,
        double progress,
        String locator,
        Long positionMillis,
        Integer pageNumber,
        Integer chapterNumber,
        boolean completed,
        long firstConsumedAtEpochMs,
        long lastConsumedAtEpochMs,
        long updatedAtEpochMs
) implements VersionedContract {
    @Override
    public String contractVersion() {
        return "v1";
    }
}

File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/ConsumptionStateV1.kt
package com.cleverferret.v2.core.database.schema

import com.cleverferret.v2.core.common.VersionedContract

data class ConsumptionStateV1(
    val consumptionStateId: String,
    val mediaItemId: String,
    val mediaVariantId: String,
    val profileId: String,
    val mode: ConsumptionModeV1,
    val progress: Double,
    val locator: String,
    val positionMillis: Long?,
    val pageNumber: Int?,
    val chapterNumber: Int?,
    val completed: Boolean,
    val firstConsumedAtEpochMs: Long,
    val lastConsumedAtEpochMs: Long,
    val updatedAtEpochMs: Long
) : VersionedContract {
    override fun contractVersion(): String = "v1"
}
